packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -730,6 +764,51 @@ class _TextSelectionGestureDetectorState extends State<TextSelectionGestureDetec
     }
   }
 
+  DragStartDetails _lastDragStartDetails;
+
+  void _handleDragStart(DragStartDetails details) {
+    assert(_lastDragStartDetails == null);
+    _lastDragStartDetails = details;
+    if (widget.onDragSelectionStart != null) {
+      widget.onDragSelectionStart(details);
+    }
+  }
+
+  Timer _dragUpdateTimer;
+  DragUpdateDetails _lastDragUpdateDetails;
+
+  void _handleDragUpdate(DragUpdateDetails details) {
+    _lastDragUpdateDetails = details;
+    // Only schedule a new timer if there's no one pending.
+    _dragUpdateTimer ??= Timer(_kDragSelectionUpdateThrottle, _handleDragUpdateThrottled);
+  }
+
+  void _handleDragUpdateThrottled() {
+    assert(_lastDragStartDetails != null);
+    assert(_lastDragUpdateDetails != null);
+    if (widget.onDragSelectionUpdate != null) {
+      widget.onDragSelectionUpdate(_lastDragStartDetails, _lastDragUpdateDetails);
+    }
+    _dragUpdateTimer = null;
```

Comment:
Maybe add some comment around why this whole debouncing thing exists and what makes 50ms better.